site/content/blog/identify-ghost-diff-during-kc-reconcile.md

```diff
@@ -0,0 +1,274 @@
+---
+title: "Identify ghost diff during Kapp Controller reconciliation"
+slug: identify-ghost-diff-during-kapp-controller-reconciliation
+date: 2022-04-21
+author: Rohit Aggarwal
+excerpt: "Identify ghost diff during Kapp Controller reconciliation "
+image: /img/logo.svg
+tags: ['carvel', 'kapp-controller', 'gitops', 'diffs', 'diff']
+---
+
+# Identify ghost diff during Kapp Controller reconciliation
+
+Kapp controller, a Package manager, is compatible with Gitops philosophy. It is continuously ensuring that the cluster is or converging towards the desired state. It does so by running the reconciliation loop after every `SyncPeriod` duration. In each reconciliation cycle, it monitors the current state of the resources on the cluster and tries to bring it to the desired state if there is any mismatch. It does so with the help of Kapp. Kapp tries to do a diff of the current live state of the resources with the desired state. Sometimes, even though there is no change in the desired state by the user explicitly, Kapp still thinks some resources have changed and tries to redeploy them as part of the reconciliation. We call them `ghost` diffs. 
```

Comment:
I think it will be "kapp" and not "Kapp" everywhere

Reply:
Adopted